### PR TITLE
add new expression Add() and Neg()

### PR DESCRIPTION
**Pull request by cturbelin**
Add 2 new study expressions : Add(...) to add numeric values and neg(value) to invert the sign of a numeric value

### CODE REVIEW

**Comment by phev8** [comment]
thank you for the PR @cturbelin 
I was thinking on the naming, and I would really prefer calling it `sum` instead, as I think it would fit better to future methods we might add (some basic statistical expressions e.g. mode, std). 

Would it be possible to quickly rename it? Or any strong feeling against it?

**Comment by cturbelin** [comment]
Nothing against it ! I do it immediately

**Comment by phev8** [comment]
What do you think about making the "sum" error tolerant, in a sense:
- skip (continue)  any of the arguments with error (don't add anything to the sum)
- but don't early return an error

Basically as if it would ignore "NaN" (and only sum valid numbers). Otherwise, you might need to put a lot of logic around to check if all items exists.

**Comment by cturbelin** [comment]
Yes, I agree the error propagation is a problem. I can do that too ! 
By the way, it could be interesting to introduce something like a debug feature, for example to have a optional component to be informed about result of each step and or collect eventual masked errors or warning to be able to properly debug these cases. To be discussed !

**Comment by cturbelin** [comment]
The changes has been done in case-editor-tools too !

**Comment by cturbelin** [comment]
@phev8 In this case should we instead be more explicit, like using a default() operator allowing to explicitly ignore an error for a given argument, with eventual default value? So it would not mask error where it could be expected ?

**Comment by phev8** [comment]
In my experience, you would never expect it to return with an error. We had that with "or" and "and" before which was very confusing.
The error is only practical for debugging, so printing it is useful.
I would say it's better in this case to have a default behaviour for "not a number" (don't add anything to sum and print for debugging until we don't have other methods for that)

Independently from this we can also add a "default" (retrieve or use defined value). But I would not want to wrap each and every dynamic item for this to avoid an error (sum being zero even if 10 of 11 values are available and I only care if sum is aber 5 or so)

**Comment by phev8** [comment]
In my experience, you would never expect it to return with an error. We had that with "or" and "and" before which was very confusing.
The error is only practical for debugging, so printing it is useful.
I would say it's better in this case to have a default behaviour for "not a number" (don't add anything to sum and print for debugging until we don't have other methods for that)

Independently from this we can also add a "default" (retrieve or use defined value). But I would not want to wrap each and every dynamic item for this to avoid an error (sum being zero even if 10 of 11 values are available and I only care if sum is aber 5 or so)

**Comment by cturbelin** [comment]
Well, I would not print it as default because if it raise a log each time a survey is submitted it can be very heavy for the logs. Unless we have a flag to enble it only in some context (like if we are running in debug mode or an expression evaluator outside the survey engine itself)

**Comment by cturbelin** [comment]
I can also add the boolean casting, adding 1 when it's true, allowing to count true conditions

**Comment by phev8** [comment]
Could be useful 👍 
There is always a slight danger where people unintentionally mix Boolean expressions and numbers but at end the logic must be tested anyways to see if works as expected by the study team.

File: pkg/studyengine/expressions.go [hunk 1]
```diff
@@ -1175,6 +1183,44 @@ func (ctx EvalContext) not(exp types.Expression) (val bool, err error) {
...
+			logger.Debug.Printf("exp %d in 'sum' returned error: %v", idx, err)
+			continue
+		}
+		switch v := arg.(type) {
+			case bool:
+				if(v) {
+					t = t + 1
+				}
+			case float64:
+				t = t + v
```
Comment (phev8) [inline pkg/studyengine/expressions.go:1200]: @cturbelin looks good to me so far, maybe one more debug print here, for the default switch case (with unexpected type at index)?

**Pull request merged**